### ISSUE TITLE
Failed to start GRC

### ISSUE THREAD

**Issue by fungaren** [gnuradio/gnuradio#4891]
After compiled gnuradio (branch `maint-3.9`), I failed to start grc, it show me a message, and the program exited:

> Failed to initialize GTK. If you are running over ssh......

Then I edit `/usr/local/bin/gnuradio-companion`:

```py
def check_gtk():
    try:
        # ========REMOVE THIS LINE==========
        #warnings.filterwarnings("error")
        import gi
        gi.require_version('Gtk', '3.0')
        gi.require_version('PangoCairo', '1.0')
        gi.require_foreign('cairo', 'Context')

        from gi.repository import Gtk
        Gtk.init_check()
        warnings.filterwarnings("always")
    except Exception as err:
        die(err, "Failed to initialize GTK. If you are running over ssh, "
                 "did you enable X forwarding and start ssh with -X?")
```

And it works, the output:

```
/usr/lib/python3/dist-packages/gi/overrides/Gdk.py:461: Warning: g_value_set_boxed: assertion 'G_VALUE_HOLDS_BOXED (value)' failed
  initialized, argv = Gdk.init_check(sys.argv)
<<< Welcome to GNU Radio Companion v3.9.2.0-50-g1a4bbabb >>>

Block paths:
        /home/fang/.grc_gnuradio
        /usr/local/share/gnuradio/grc/blocks
```

I think i missed something. Dose anyone know it?

**Comment by dkozel** [comment]
Hi @mooction

What operating system are you compiling on?

**Comment by mooction** [comment]
@dkozel Deepin 20.2.1 (upstream Debian version 10.9)

`cmake -L ../`

```text
-- Build type set to Release.
-- Build date set to Fri, 16 Jul 2021 01:44:22Z.
-- 
-- The build system will automatically enable all components.
-- Use -DENABLE_DEFAULT=OFF to disable components by default.
-- Boost version: 1.67.0
-- Found the following Boost libraries:
--   date_time
--   program_options
--   filesystem
--   system
--   regex
--   thread
--   unit_test_framework
--   chrono
--   atomic
-- 
-- Configuring testing-support support...
--   Dependency Boost_UNIT_TEST_FRAMEWORK_FOUND = ON
--   Enabling testing-support support.
--   Override with -DENABLE_TESTING=ON/OFF
-- Extracting version information from git describe...
-- Compiler Version: cc (Uos 8.3.0.3-3+rebuild) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-- Compiler Flags: /usr/bin/cc:::-O3 -DNDEBUG  -fvisibility=hidden -Wsign-compare -Wall -Wno-uninitialized
/usr/bin/c++:::-O3 -DNDEBUG  -fvisibility=hidden -Wsign-compare -Wall -Wno-uninitialized
-- ADDING PERF COUNTERS
-- User set python executable /usr/bin/python3
-- Python checking for numpy - found
-- Python checking for pygccxml - not found
-- Found pybind11: /usr/local/include (found version "2.6.3" dev1)
-- 
-- Configuring python-support support...
--   Dependency PYTHONLIBS_FOUND = TRUE
--   Dependency pybind11_FOUND = 1
--   Dependency NUMPY_FOUND = TRUE
--   Enabling python-support support.
--   Override with -DENABLE_PYTHON=ON/OFF
-- 
-- Configuring VOLK support...
--   Found VOLK:
--   * Version: 2.5.0
--   * Libraries: Volk::volk
--   * Includes: /usr/local/include
-- Not using additional GNU Radio native architecture optimizations.
-- Checking for module 'gmp'
--   No package 'gmp' found
-- Using GMP.
-- 
-- Configuring post-install support...
--   Enabling post-install support.
--   Override with -DENABLE_POSTINSTALL=ON/OFF
-- Could NOT find MathJax2 (missing: MATHJAX2_JS_PATH) 
-- 
-- Configuring doxygen support...
--   Dependency DOXYGEN_FOUND = YES
--   Enabling doxygen support.
--   Override with -DENABLE_DOXYGEN=ON/OFF
CMake Warning at docs/doxygen/CMakeLists.txt:24 (message):
  MathJax 2 not found, HTML equations might not be properly rendered.


-- Checking for module 'gmp'
--   No package 'gmp' found
-- Using GMP.
-- 
-- Configuring gnuradio-runtime support...
--   Dependency Boost_FOUND = 1
--   Dependency PYTHONINTERP_FOUND = TRUE
--   Dependency MPLIB_FOUND = TRUE
--   Dependency LOG4CPP_FOUND = TRUE
--   Dependency pybind11_FOUND = 1
--   Enabling gnuradio-runtime support.
--   Override with -DENABLE_GNURADIO_RUNTIME=ON/OFF
-- 
-- Configuring gr-ctrlport support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Enabling gr-ctrlport support.
--   Override with -DENABLE_GR_CTRLPORT=ON/OFF
-- Checking for module 'thrift'
--   No package 'thrift' found
-- thrift looking for version 0.9.2
-- Binary 'thrift' not found.
-- Checking for module 'libunwind'
--   No package 'libunwind' found
-- Loading build date Fri, 16 Jul 2021 01:44:22 into constants...
-- Loading version v3.9.2.0-50-g1a4bbabb into constants...
-- TRY_SHM_VMCIRCBUF set to ON.
-- 
-- Python checking for PyYAML >= 3.11 - found
-- Python checking for mako >= 1.0.7 - found
-- Python checking for pygobject >= 2.28.6 - found
-- Python checking for Gtk (GI) >= 3.10.8 - found
-- Python checking for Cairo (GI) >= 1.0 - found
-- Python checking for PangoCairo (GI) >= 1.0 - found
-- Python checking for numpy - found
-- 
-- Configuring gnuradio-companion support...
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_PYTHON = ON
--   Dependency PYYAML_FOUND = TRUE
--   Dependency MAKO_FOUND = TRUE
--   Dependency PYGI_FOUND = TRUE
--   Dependency GTK_GI_FOUND = TRUE
--   Dependency CAIRO_GI_FOUND = TRUE
--   Dependency PANGOCAIRO_GI_FOUND = TRUE
--   Dependency NUMPY_FOUND = TRUE
--   Enabling gnuradio-companion support.
--   Override with -DENABLE_GRC=ON/OFF
-- Python checking for pytest - found
-- 
-- Configuring gr-blocks support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Enabling gr-blocks support.
--   Override with -DENABLE_GR_BLOCKS=ON/OFF
-- 
-- Configuring gr-fec support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Enabling gr-fec support.
--   Override with -DENABLE_GR_FEC=ON/OFF
-- 
-- Configuring gr-fft support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency FFTW3f_FOUND = TRUE
--   Enabling gr-fft support.
--   Override with -DENABLE_GR_FFT=ON/OFF
-- 
-- Configuring gr-filter support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_FFT = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Enabling gr-filter support.
--   Override with -DENABLE_GR_FILTER=ON/OFF
-- Python checking for pyqtgraph - not found
-- Python checking for scipy - found
CMake Warning at gr-filter/CMakeLists.txt:50 (message):
  PyQtGraph and Scipy are required to run the filter design tool, but are not
  detected! Please make sure they are installed on the target system.


-- 
-- Configuring gr-analog support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency ENABLE_GR_FFT = ON
--   Dependency ENABLE_GR_FILTER = ON
--   Enabling gr-analog support.
--   Override with -DENABLE_GR_ANALOG=ON/OFF
-- 
-- Configuring gr-digital support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_FFT = ON
--   Dependency ENABLE_GR_FILTER = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency ENABLE_GR_ANALOG = ON
--   Enabling gr-digital support.
--   Override with -DENABLE_GR_DIGITAL=ON/OFF
-- 
-- Configuring gr-dtv support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_ANALOG = ON
--   Dependency ENABLE_GR_FILTER = ON
--   Dependency ENABLE_GR_FEC = ON
--   Enabling gr-dtv support.
--   Override with -DENABLE_GR_DTV=ON/OFF
-- 
-- Configuring gr-audio support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Enabling gr-audio support.
--   Override with -DENABLE_GR_AUDIO=ON/OFF
-- Found ALSA 1.1.8
-- Checking for module 'jack'
--   No package 'jack' found
-- Checking for module 'portaudio-2.0'
--   No package 'portaudio-2.0' found
-- 
-- Configuring gr-channels support...
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency ENABLE_GR_FFT = ON
--   Dependency ENABLE_GR_FILTER = ON
--   Dependency ENABLE_GR_ANALOG = ON
--   Enabling gr-channels support.
--   Override with -DENABLE_GR_CHANNELS=ON/OFF
-- Python checking for PyQt5 - found
-- Checking for module 'Qt5Qwt6'
--   No package 'Qt5Qwt6' found
-- QWT Version: 6.1.4
-- 
-- Configuring gr-qtgui support...
--   Dependency Boost_FOUND = 1
--   Dependency QT_FOUND = 1
--   Dependency QWT_FOUND = TRUE
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_FFT = ON
--   Dependency ENABLE_GR_FILTER = ON
--   Dependency PYTHONLIBS_FOUND = TRUE
--   Dependency PYQT5_FOUND = TRUE
--   Enabling gr-qtgui support.
--   Override with -DENABLE_GR_QTGUI=ON/OFF
-- 
-- Configuring gr-trellis support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_ANALOG = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency ENABLE_GR_DIGITAL = ON
--   Enabling gr-trellis support.
--   Override with -DENABLE_GR_TRELLIS=ON/OFF
-- 
-- Configuring gr-uhd support...
--   Dependency Boost_FOUND = 1
--   Dependency UHD_FOUND = TRUE
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_FILTER = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency ENABLE_GR_ANALOG = ON
--   Enabling gr-uhd support.
--   Override with -DENABLE_GR_UHD=ON/OFF
-- 
-- Configuring gr-uhd UHD 4.0 RFNoC support...
--   Dependency ENABLE_GR_UHD = ON
--   Dependency UHD_FOUR_POINT_OH_RFNOC = FALSE
--   Disabling gr-uhd UHD 4.0 RFNoC support.
--   Override with -DENABLE_UHD_RFNOC=ON/OFF
--   UHD Version: 3.13.1.0-3
-- Python checking for Mako >= 1.0.7 - found
-- Python checking for click - found
-- Python checking for click-plugins - found
-- 
-- Configuring gr-utils support...
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_PYTHON = ON
--   Dependency MAKO_FOUND = TRUE
--   Enabling gr-utils support.
--   Override with -DENABLE_GR_UTILS=ON/OFF
-- 
-- Configuring gr_modtool support...
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_PYTHON = ON
--   Dependency CLICK_FOUND = TRUE
--   Dependency CLICK_PLUGINS_FOUND = TRUE
--   Enabling gr_modtool support.
--   Override with -DENABLE_GR_MODTOOL=ON/OFF
-- 
-- Configuring gr_blocktool support...
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_PYTHON = ON
--   Enabling gr_blocktool support.
--   Override with -DENABLE_GR_BLOCKTOOL=ON/OFF
-- 
-- Configuring gr-video-sdl support...
--   Dependency SDL_FOUND = TRUE
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Enabling gr-video-sdl support.
--   Override with -DENABLE_GR_VIDEO_SDL=ON/OFF
-- 
-- Configuring gr-vocoder support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_FFT = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency ENABLE_GR_FILTER = ON
--   Dependency ENABLE_GR_ANALOG = ON
--   Enabling gr-vocoder support.
--   Override with -DENABLE_GR_VOCODER=ON/OFF
-- 
-- Configuring gr-wavelet support...
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ENABLE_GR_BLOCKS = ON
--   Dependency ENABLE_GR_ANALOG = ON
--   Dependency GSL_FOUND = TRUE
--   Enabling gr-wavelet support.
--   Override with -DENABLE_GR_WAVELET=ON/OFF
-- 
-- Configuring gr-zeromq support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Dependency ZEROMQ_FOUND = TRUE
--   Enabling gr-zeromq support.
--   Override with -DENABLE_GR_ZEROMQ=ON/OFF
-- 
-- Configuring gr-network support...
--   Dependency Boost_FOUND = 1
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Enabling gr-network support.
--   Override with -DENABLE_GR_NETWORK=ON/OFF
CMake Warning at gr-soapy/CMakeLists.txt:16 (find_package):
  By not providing "FindSoapySDR.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "SoapySDR",
  but CMake did not find one.

  Could not find a package configuration file provided by "SoapySDR"
  (requested version 0.7.2) with any of the following names:

    SoapySDRConfig.cmake
    soapysdr-config.cmake

  Add the installation prefix of "SoapySDR" to CMAKE_PREFIX_PATH or set
  "SoapySDR_DIR" to a directory containing one of the above files.  If
  "SoapySDR" provides a separate development package or SDK, be sure it has
  been installed.


-- 
-- Configuring gr-soapy support...
--   Dependency Boost_FOUND = 1
--   Dependency SoapySDR_FOUND = 0
--   Dependency ENABLE_GNURADIO_RUNTIME = ON
--   Disabling gr-soapy support.
--   Override with -DENABLE_GR_SOAPY=ON/OFF
-- 
-- ######################################################
-- # Gnuradio enabled components                         
-- ######################################################
--   * testing-support
--   * python-support
--   * post-install
--   * doxygen
--   * gnuradio-runtime
--   * gr-ctrlport
--   * gnuradio-companion
--   * gr-blocks
--   * gr-fec
--   * gr-fft
--   * gr-filter
--   * gr-analog
--   * gr-digital
--   * gr-dtv
--   * gr-audio
--   * * alsa
--   * * oss
--   * gr-channels
--   * gr-qtgui
--   * gr-trellis
--   * gr-uhd
--   * gr-utils
--   * gr_modtool
--   * gr_blocktool
--   * gr-video-sdl
--   * gr-vocoder
--   * * codec2
--   * * freedv
--   * * gsm
--   * gr-wavelet
--   * gr-zeromq
--   * gr-network
-- 
-- ######################################################
-- # Gnuradio disabled components                        
-- ######################################################
--   * gr-uhd UHD 4.0 RFNoC
--   * gr-soapy
-- 
-- Using install prefix: /usr/local
-- Building for version: v3.9.2.0-50-g1a4bbabb / 3.9.2git
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fang/gnuradio/build
-- Cache values
APPSTREAMCLI:FILEPATH=APPSTREAMCLI-NOTFOUND
BUILD_SHARED_LIBS:BOOL=ON
CMAKE_BUILD_TYPE:STRING=Release
CMAKE_INSTALL_PREFIX:PATH=/usr/local
ENABLE_BAD_BOOST:BOOL=OFF
ENABLE_CTRLPORT_THRIFT:BOOL=OFF
ENABLE_DOXYGEN:BOOL=ON
ENABLE_EXAMPLES:BOOL=ON
ENABLE_GNURADIO_RUNTIME:BOOL=ON
ENABLE_GRC:BOOL=ON
ENABLE_GR_ANALOG:BOOL=ON
ENABLE_GR_AUDIO:BOOL=ON
ENABLE_GR_BLOCKS:BOOL=ON
ENABLE_GR_BLOCKTOOL:BOOL=ON
ENABLE_GR_CHANNELS:BOOL=ON
ENABLE_GR_CTRLPORT:BOOL=ON
ENABLE_GR_DIGITAL:BOOL=ON
ENABLE_GR_DTV:BOOL=ON
ENABLE_GR_FEC:BOOL=ON
ENABLE_GR_FFT:BOOL=ON
ENABLE_GR_FILTER:BOOL=ON
ENABLE_GR_MODTOOL:BOOL=ON
ENABLE_GR_NETWORK:BOOL=ON
ENABLE_GR_QTGUI:BOOL=ON
ENABLE_GR_TRELLIS:BOOL=ON
ENABLE_GR_UHD:BOOL=ON
ENABLE_GR_UTILS:BOOL=ON
ENABLE_GR_VIDEO_SDL:BOOL=ON
ENABLE_GR_VOCODER:BOOL=ON
ENABLE_GR_WAVELET:BOOL=ON
ENABLE_GR_ZEROMQ:BOOL=ON
ENABLE_NATIVE:BOOL=OFF
ENABLE_PERFORMANCE_COUNTERS:BOOL=ON
ENABLE_POSTINSTALL:BOOL=ON
ENABLE_PYTHON:BOOL=ON
ENABLE_TESTING:BOOL=ON
GRC_BLOCKS_DIR:PATH=share/gnuradio/grc/blocks
GRC_XTERM_EXE:FILEPATH=GRC_XTERM_EXE-NOTFOUND
GR_CONF_DIR:PATH=etc
GR_DATA_DIR:PATH=share
GR_DOC_DIR:PATH=share/doc
GR_INCLUDE_DIR:PATH=include
GR_LIBEXEC_DIR:PATH=libexec
GR_LIBRARY_DIR:PATH=lib
GR_PKG_CONF_DIR:PATH=/usr/local/etc/gnuradio/conf.d
GR_PKG_DATA_DIR:PATH=share/gnuradio
GR_PKG_DOC_DIR:PATH=share/doc/gnuradio-3.9.2git
GR_PKG_LIBEXEC_DIR:PATH=libexec/gnuradio
GR_PREFSDIR:PATH=/usr/local/etc/gnuradio/conf.d
GR_RUNTIME_DIR:PATH=bin
GR_THEMES_DIR:PATH=share/gnuradio/themes
HAVE_XDG_UTILS:FILEPATH=/usr/bin/xdg-desktop-menu
JACK_INCLUDE_DIR:PATH=JACK_INCLUDE_DIR-NOTFOUND
JACK_LIBRARY:FILEPATH=JACK_LIBRARY-NOTFOUND
LIB_SUFFIX:STRING=
LOG4CPP_INCLUDE_DIR:PATH=/usr/include
LOG4CPP_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/liblog4cpp.so
PYBIND11_PYTHON_VERSION:STRING=
QA_PYTHON_EXECUTABLE:FILEPATH=/usr/bin/python3
Qt5Core_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5Core
Qt5Gui_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5Gui
Qt5Widgets_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/Qt5Widgets
SDLMAIN_LIBRARY:FILEPATH=/usr/lib/x86_64-linux-gnu/libSDLmain.a
SDL_INCLUDE_DIR:PATH=/usr/include/SDL
SDL_LIBRARY:STRING=/usr/lib/x86_64-linux-gnu/libSDLmain.a;/usr/lib/x86_64-linux-gnu/libSDL.so;-lpthread
SHELL:FILEPATH=/usr/bin/sh
SYSCONFDIR:PATH=/usr/local/etc
SoapySDR_DIR:PATH=SoapySDR_DIR-NOTFOUND
THRIFT_BIN:FILEPATH=THRIFT_BIN-NOTFOUND
THRIFT_INCLUDE_DIRS:PATH=THRIFT_INCLUDE_DIRS-NOTFOUND
THRIFT_LIBRARIES:FILEPATH=THRIFT_LIBRARIES-NOTFOUND
TRY_SHM_VMCIRCBUF:BOOL=ON
UHD_DIR:PATH=/usr/lib/x86_64-linux-gnu/cmake/uhd
VOLK_FOUND:BOOL=TRUE
Volk_DIR:PATH=/usr/local/lib/cmake/volk
XMLTO_EXECUTABLE:FILEPATH=XMLTO_EXECUTABLE-NOTFOUND
libunwind_INCLUDE_DIR:PATH=libunwind_INCLUDE_DIR-NOTFOUND
libunwind_LIBRARY:FILEPATH=libunwind_LIBRARY-NOTFOUND
pkgcfg_lib_ALSA_PKGCONF_asound:FILEPATH=/usr/lib/x86_64-linux-gnu/libasound.so
pkgcfg_lib_GSL_gsl:FILEPATH=/usr/lib/x86_64-linux-gnu/libgsl.so
pkgcfg_lib_GSL_gslcblas:FILEPATH=/usr/lib/x86_64-linux-gnu/libgslcblas.so
pkgcfg_lib_GSL_m:FILEPATH=/usr/lib/x86_64-linux-gnu/libm.so
pkgcfg_lib_LIBCODEC2_PKG_codec2:FILEPATH=/usr/lib/x86_64-linux-gnu/libcodec2.so
pkgcfg_lib_PC_FFTW3f_fftw3f:FILEPATH=/usr/lib/x86_64-linux-gnu/libfftw3f.so
pkgcfg_lib_PC_SNDFILE_sndfile:FILEPATH=/usr/lib/x86_64-linux-gnu/libsndfile.so
pkgcfg_lib_PC_UHD_boost_system:FILEPATH=/usr/lib/x86_64-linux-gnu/libboost_system.so
pkgcfg_lib_PC_UHD_uhd:FILEPATH=/usr/lib/x86_64-linux-gnu/libuhd.so
pkgcfg_lib_PC_ZEROMQ_zmq:FILEPATH=/usr/lib/x86_64-linux-gnu/libzmq.so
pybind11_DIR:PATH=/usr/local/share/cmake/pybind11
```

**Comment by marcusmueller** [comment]
So, there's two things at work here:

1. GTK/GDK on your machine is not happy with itself. In the `Gtk.init_check()` line, it does check whether its internal typing system is in good order, and find it should be emitting a warning. So, that's not something that GNU Radio can fix!
2. GNU Radio might be a bit overly sensitive here: A warning is a warning, not something fatal. But I **bet** that conversion of warnings to exceptions (which you commented out) is in there for a reason! So we need to figure out that reason.

**Comment by marcusmueller** [comment]
@mooction can you change things to

```python
def check_gtk():
    try:
        # ========CHANGE THIS LINE==========
        warnings.filterwarnings("always")
        import gi
        gi.require_version('Gtk', '3.0')
        gi.require_version('PangoCairo', '1.0')
        gi.require_foreign('cairo', 'Context')

        from gi.repository import Gtk
        if not Gtk.init_check():
            warnings.warn("init_check unsuccessful. We're in a bad state. Every usage of Gtk after this point has undefined behaviour!\nhttps://developer.gnome.org/gtk3/stable/gtk3-General.html#gtk-init-check")
        warnings.filterwarnings("always")
    except Exception as err:
        die(err, "Failed to initialize GTK. If you are running over ssh, "
                 "did you enable X forwarding and start ssh with -X?")
```

and tell us the console output of starting gnuradio-companion?

**Comment by mooction** [comment]
@marcusmueller I used the code, and GRC works. The output is the same:

```text
/usr/lib/python3/dist-packages/gi/overrides/Gdk.py:461: Warning: g_value_set_boxed: assertion 'G_VALUE_HOLDS_BOXED (value)' failed
  initialized, argv = Gdk.init_check(sys.argv)
<<< Welcome to GNU Radio Companion v3.9.2.0-50-g1a4bbabb >>>

Block paths:
        /home/fang/.grc_gnuradio
        /usr/local/share/gnuradio/grc/blocks
```

Looks not so bad?

**Comment by mormj** [comment]
@marcusmueller - can you verify that this can be closed based on #5216 and #5217

**Comment by marcusmueller** [comment]
Yep, seems done.
